neural_network/back_propagation_neural_network.py

```diff
@@ -21,7 +21,7 @@
 from matplotlib import pyplot as plt
 
 
-def sigmoid(x):
+def sigmoid(x) -> float:
```

Comment:
```suggestion
def sigmoid(x: np.ndarray) -> np.ndarray:
```
The sigmoid function is being computed element-wise on numpy arrays in this file